scipy/integrate/_ivp/common.py

```diff
@@ -108,7 +111,12 @@ def select_initial_step(fun, t0, y0, f0, direction, order, rtol, atol):
         h0 = 1e-6
     else:
         h0 = 0.01 * d0 / d1
-
+    if t_bound is not None:
+        # Check t0+h0*direction doesn't take us beyond t_bound
+        if h0 > (t_bound-t0)/direction: 
+            h0 = (t_bound-t0)/direction
```

Comment:
Nit: the logic `h = min(h, (t_bound-t0)/direction)` is used below. Consider using the same here idea here.

Reply:
Fixed.